### PR TITLE
Uri fixes

### LINKED ISSUE #3197: Add a constructor for Uri that segments the given path
Right now I'm building a Uri and I have this:
```scala
def url(tenantId: TenantId) = 
  (uri"https://manage.office.com" / s"api/v1.0/${tenantId.string}/activity/feed/subscriptions/content")
    .withQueryParam("contentType", "Audit.AzureActiveDirectory")
    .withQueryParam("startTime", "2020-02-08T11:50")
    .withQueryParam("endTime", "2020-02-08T12:00")
```
It would be nice if I could do a builder-like pattern for the whole thing like so:
```scala
def url(tenantId: TenantId) =
  uri"https://manage.office.com"
    .addPath(s"api/v1.0/${tenantId.string}/activity/feed/subscriptions/content")
    .withQueryParam("contentType", "Audit.AzureActiveDirectory")
    .withQueryParam("startTime", "2020-02-08T11:50")
    .withQueryParam("endTime", "2020-02-08T12:00")
```

There is currently `withPath`, but it doesn't URL encode and I'm told it's meant to be a wrapper around the copy constructor.

It should be very easy to add.

### PR DESCRIPTION
**Pull request by hamnis**
Added scaladoc to Uri.withPath to describe behaviour
Be nice and add a / if the path does not have that and is non-empty.

Fixes #3197

### CODE REVIEW

File: core/src/main/scala/org/http4s/Uri.scala [hunk 1]
```diff
@@ -105,7 +111,14 @@ final case class Uri(
 
       case Uri(None, None, _, _, _) =>
     }
-    writer << path
+
+    this match {
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/Uri.scala]: What if there is a scheme and no authority?  I guess this behavior is scheme specific?

File: core/src/main/scala/org/http4s/Uri.scala [hunk 2]
```diff
@@ -35,20 +35,26 @@ final case class Uri(
...
-  def /(newSegment: Path): Uri = {
+  def addPath(newSegment: Path): Uri = {
     val encoded = pathEncode(newSegment)
     val newPath =
       if (path.isEmpty || path.last != '/') s"$path/$encoded"
       else s"$path$encoded"
     copy(path = newPath)
   }
 
+  def /(newSegment: Path): Uri = addPath(newSegment)
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/Uri.scala]: We could document this as an alias.

File: core/src/main/scala/org/http4s/Uri.scala [hunk 3]
```diff
@@ -35,20 +35,26 @@ final case class Uri(
...
+    * @param path the path string to replace
+    */
   def withPath(path: Path): Uri = copy(path = path)
 
   def withFragment(fragment: Fragment): Uri = copy(fragment = Option(fragment))
 
   def withoutFragment: Uri = copy(fragment = Option.empty[Fragment])
 
-  def /(newSegment: Path): Uri = {
+  def addPath(newSegment: Path): Uri = {
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/Uri.scala]: I like having a non-symbolic name.

**Comment by hamnis** [comment]
> Any reason this can't be on series/0.20? I've been trying to backport bug fixes until we officially EOL it.
> 
> We talked about a short, but indefinite, wind down, but should put a date on it.

No, no reason. i can move it there.

File: core/src/main/scala/org/http4s/Uri.scala (hunk 1 above)
Comment (hamnis) [reply core/src/main/scala/org/http4s/Uri.scala]: I do believe you need an authority for the path segmenting to make sense.

File: core/src/main/scala/org/http4s/Uri.scala (hunk 2 above)
Comment (hamnis) [reply core/src/main/scala/org/http4s/Uri.scala]: yes

File: core/src/main/scala/org/http4s/Uri.scala (hunk 3 above)
Comment (hamnis) [reply core/src/main/scala/org/http4s/Uri.scala]: i'm renaming this as `addSegment`

File: core/src/main/scala/org/http4s/Uri.scala (hunk 1 above)
Comment (rossabaker) [reply core/src/main/scala/org/http4s/Uri.scala]: `file://` is the one that jumped to my mind.  But not everything has a `/`-rooted filesystem.

**Pull request merged**